src/fusor/fusor.py

```diff
@@ -266,55 +264,53 @@ async def transcript_segment_element(
                 )
                 _logger.warning(msg)
                 return None, [msg]
-            residue_mode = kwargs.get("residue_mode")
-            # TODO: Remove once fixed in cool_seq_tool - need to verify that this code isn't still needed
-            if residue_mode != ResidueMode.INTER_RESIDUE:
-                start = kwargs.get("start")
-                kwargs["start"] = start - 1 if start is not None else None
-                kwargs["residue_mode"] = "inter-residue"
             chromosome = kwargs.get("chromosome")
-            # if chromosome is a string, assume it's an accession, fix it for the kwargs since CST expects this as alt_ac
+            # if chromosome is a string, assume it's an accession, fix it for the kwargs since CST expects this as genomic_ac
             if type(chromosome) is str:
-                kwargs["alt_ac"] = chromosome
-            data = await self.cool_seq_tool.ex_g_coords_mapper.genomic_to_transcript_exon_coordinates(
+                kwargs["genomic_ac"] = chromosome
+            data = await self.cool_seq_tool.ex_g_coords_mapper.genomic_to_tx_segment(
                 **kwargs
             )
 
-        if data.genomic_data is None:
-            return None, data.warnings
-
-        genomic_data = data.genomic_data
-        genomic_data.transcript = coerce_namespace(genomic_data.transcript)
+        data.tx_ac = coerce_namespace(data.tx_ac)
 
         normalized_gene_response = self._normalized_gene(
-            genomic_data.gene, use_minimal_gene=use_minimal_gene
+            data.gene, use_minimal_gene=use_minimal_gene
         )
         if not normalized_gene_response[0] and normalized_gene_response[1]:
             return None, [normalized_gene_response[1]]
 
+        seg_start = data.seg_start
+        genomic_start_location = seg_start.genomic_location if seg_start else None
+
+        seg_end = data.seg_end
+        genomic_end_location = seg_end.genomic_location if seg_end else None
+
         return (
             TranscriptSegmentElement(
-                transcript=genomic_data.transcript,
-                exonStart=genomic_data.exon_start,
-                exonStartOffset=genomic_data.exon_start_offset,
-                exonEnd=genomic_data.exon_end,
-                exonEndOffset=genomic_data.exon_end_offset,
+                transcript=data.tx_ac,
+                # offset by 1 because in CST exons are 0-based
+                exonStart=seg_start.exon_ord + 1 if seg_start else None,
+                exonStartOffset=seg_start.offset if seg_start else None,
+                # offset by 1 because in CST exons are 0-based
+                exonEnd=seg_end.exon_ord + 1 if seg_end else None,
+                exonEndOffset=seg_end.offset if seg_end else None,
                 gene=normalized_gene_response[0],
                 elementGenomicStart=self._sequence_location(
```

Comment:
I think this method should be modified. Since Cool-Seq-Tool already constructs the SequenceLocation, I believe you only need to run ga4gh identify

Reply:
yeah, at first I had an implementation where we just returned the SequenceLocation as returned from cool-seq-tool but ended up changing ti back to this since we needed the id's to be a bit different. I think I may just change this to use the location returned from CST but modify the id's to be what we want (the ga4gh identified one and the genomic_ac for the id for the sequenceReference)

Reply:
@korikuzma fixed this in the latest commit! I refactored some stuff and created a function to add these id's as we expect them in fusor